Stack/Opc.Ua.Core/Stack/Server/ServerBase.cs

```diff
@@ -805,6 +845,16 @@ protected virtual void OnCertificateUpdate(object sender, CertificateUpdateEvent
                 }
             }
 
+            //update certificate in the endpoint descriptions
+            foreach (EndpointDescription endpointDescription in m_endpoints)
+            {
+                SetServerCertificateInEndpointDescription(
```

Comment:
this calculates the ServerCertificate for every endpoint description. For the time being it would be better to split the calculation of the `ServerCertificate` blob to a separate function, then use the same blob for every description, as needed by the security profile.